### PR TITLE
feat: new product gpd_for_aca PAGOPA-2195

### PR DESCRIPTION
**Pull request by FedericoRuzzier**
### Commands

Domain gps: `sh terraform.sh apply weu-dev -target=module.apim_api_debt_positions_api_v1 -target=module.apim_api_debt_positions_api_v2 -target=module.apim_api_gpd_api -target=module.apim_api_gpd_api_v2 -target=azurerm_api_management_api_operation_policy.set_service_type_on_create_v1 -target=terraform_data.sha256_set_service_type_on_create_v1`

Domain aca: `sh terraform.sh apply weu-dev -target=azurerm_api_management_api_version_set.api_debt_positions_for_aca_api -target=module.apim_api_debt_positions_for_aca_api_v1`

```
sh terraform.sh apply  weu-dev --target=azapi_resource.service_type_set_fragment --target=terraform_data.sha256_service_type_set_fragment
```
```
sh terraform.sh apply weu-dev --target=azapi_resource.gpd_external_policy_fragment --target=terraform_data.sha256_gpd_external_policy_fragment
```

```
sh terraform.sh apply weu-dev --target=azurerm_api_management_api_version_set.api_debt_positions_api --target=module.apim_api_debt_positions_api_v1 --target=module.apim_api_debt_positions_api_v2
```
```
sh terraform.sh apply weu-dev --target=terraform_data.sha256_segregation_codes_fragment --target=azapi_resource.segregation_codes_fragment
```

### List of changes
New product gpd_for_aca with aca apis

### Motivation and context
Needed in order to extract subscription keys

### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### CODE REVIEW

File: src/domains/aca-app/api/gpd-for-aca-services/v1/_base_policy.xml [hunk 1]
```diff
@@ -0,0 +1,81 @@
+<policies>
+  <inbound>
+    <choose>
+      <when condition="@(Regex.IsMatch(context.Request.Url.Path, @"^\/pagopa-gpd-core\/organizations\/([a-zA-Z0-9]{11})\/debtpositions$", RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(500)) && ((string)(context.Request.Method)).Equals("POST"))">
```
Comment (aomegax) [inline src/domains/aca-app/api/gpd-for-aca-services/v1/_base_policy.xml:4]: move from all-api to single api (for example https://github.com/pagopa/pagopa-infra/blob/f6d39b2a9e336725cf871a43f6a9371b50e82463/src/core/apim_payment_manager.tf#L930)

File: src/domains/aca-app/api/gpd-for-aca-services/v1/_base_policy.xml [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+      remaining-calls-variable-name="remainingCallsPerSubscription" />
+    <!-- Calling Authorizer's fragment -->
+    <set-variable name="application_domain" value="gpd" />
+    <set-variable name="metadata" value="no-metadata" />
+    <choose>
+      <!-- Making sure that will excludes all APIs that does not includes CI fiscal code -->
+      <when condition="@(context.Request.MatchedParameters.ContainsKey("organizationfiscalcode") && !context.Request.Headers.GetValueOrDefault("Ocp-Apim-Subscription-Key", "").Equals(""))">
+        <set-variable name="authorization_entity" value="@(context.Request.MatchedParameters["organizationfiscalcode"])" />
+        <include-fragment fragment-id="authorizer" />
+        <!-- ################################################################### -->
```
Comment (aomegax) [inline src/domains/aca-app/api/gpd-for-aca-services/v1/_base_policy.xml:26]: This is common to GPD, please create a fragment generalizing code

File: src/domains/gps-app/api/gpd_api/debt-position-services/v1/_base_policy.xml [hunk 3]
```diff
@@ -1,5 +1,12 @@
 <policies>
   <inbound>
+    <choose>
+      <when condition="@(Regex.IsMatch(context.Request.Url.Path, @"^\/pagopa-gpd-core\/organizations\/([a-zA-Z0-9]{11})\/debtpositions$", RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(500)) && ((string)(context.Request.Method)).Equals("POST"))">
```
Comment (aomegax) [inline src/domains/gps-app/api/gpd_api/debt-position-services/v1/_base_policy.xml:4]: as before, add to the specific api

File: src/domains/gps-app/api/gpd_api/debt-position-services/v2/_base_policy.xml [hunk 4]
```diff
@@ -1,5 +1,12 @@
 <policies>
   <inbound>
+    <choose>
+      <when condition="@(Regex.IsMatch(context.Request.Url.Path, @"^\/pagopa-gpd-core\/organizations\/([a-zA-Z0-9]{11})\/debtpositions$", RegexOptions.IgnoreCase, TimeSpan.FromMilliseconds(500)) && ((string)(context.Request.Method)).Equals("POST"))">
```
Comment (aomegax) [inline src/domains/gps-app/api/gpd_api/debt-position-services/v2/_base_policy.xml:4]: as said before.
perhaps you can create a fragment too, where the input params are path and serviceType values

**Review by aomegax — CHANGES_REQUESTED** [review]

File: src/domains/gps-app/99_variables.tf [hunk 5]
```diff
@@ -442,3 +442,9 @@ variable "flag_responsetime_alert" {
   description = "Flag to enable if payments-pull response time alert is available"
   default     = 0
 }
+
+variable "service_type_gpd" {
```
Comment (FedericoRuzzier) [inline src/domains/gps-app/99_variables.tf:446]: Delete

File: src/domains/aca-app/99_variables.tf [hunk 6]
```diff
@@ -117,3 +117,9 @@ variable "tls_cert_check_helm" {
   })
   description = "tls cert helm chart configuration"
 }
+
+variable "service_type_aca" {
```
Comment (FedericoRuzzier) [inline src/domains/aca-app/99_variables.tf:121]: Delete

**Pull request merged**